api/scpca_portal/readme_file.py

```diff
@@ -12,6 +14,112 @@
 TEMPLATE_ROOT = settings.TEMPLATE_PATH / "readme"
 TEMPLATE_FILE_PATH = TEMPLATE_ROOT / "readme.md"
 
+# Dataset Readme Templates
+README_ROOT = settings.TEMPLATE_PATH / "dataset_readme"
+
+# used in get_dataset_contents_section and in 2_contents.md
+ContentRow = namedtuple("ContentRow", ["project", "modality", "format", "docs"])
+
+
+def get_dataset_contents_section(dataset):
+
+    # Metadata
+    if dataset.format == DatasetFormats.METADATA:
+        return [
+            ContentRow(project, project.modalities, "NA", "METADATA")
+            for project in dataset.projects
+        ]
+
+    rows = set()
+
+    # SINGLE_CELL
+    for project in dataset.single_cell_projects:
+        content_dict = {
+            "project": project,
+            "modality": Modalities.SINGLE_CELL,
+            "format": dataset.format,
+        }
+
+        # MERGED
+        if dataset.data[project.scpca_id].get(DatasetDataProjectConfig.MERGE_SINGLE_CELL):
+            if dataset.format == DatasetFormats.ANN_DATA:
+                content_dict["docs"] = "MERGED_ANN_DATA"
+            else:
+                content_dict["docs"] = "MERGED_SINGLE_CELL_EXPERIMENT"
+
+            rows.add(ContentRow(**content_dict))
+            continue
+
+        # MULTIPLEXED
+        if (
+            dataset.get_samples(project.scpca_id, Modalities.SINGLE_CELL)
+            and dataset.format == DatasetFormats.SINGLE_CELL_EXPERIMENT
+        ):
+            content_dict["docs"] = "MULTIPLEXED_SINGLE_CELL"
+            rows.add(ContentRow(**content_dict))
+            continue
+
+        # SINGLE_CELL
+        content_dict["docs"] = "SINGLE_CELL"
+        rows.add(ContentRow(**content_dict))
+
+    # SPATIAL
+    if dataset.format == DatasetFormats.SINGLE_CELL_EXPERIMENT:
+        for project in dataset.spatial_projects:
+            rows.add(ContentRow(project, Modalities.SPATIAL, dataset.format, "SPATIAL_LINK"))
+
+    # ANN_DATA CITE-SEQ
+    if dataset.format == DatasetFormats.ANN_DATA:
+        for project in dataset.cite_seq_projects:
+            rows.add(
+                ContentRow(
+                    project,
+                    Modalities.CITE_SEQ,
+                    dataset.format,
+                    "ANN_DATA_CITE_SEQ_LINK",
+                )
+            )
```

Comment:
This needs to be combined or replace the link when merged or when regular single-cell that is also ann_data.

Add an additional check on line 46 and 63.